beetsplug/spotify.py

```diff
@@ -195,12 +196,17 @@ def _handle_response(self, request_type, url, params=None):
                 raise SpotifyAPIError(f'API Error: {e.response.status_code}\n'
                                       f'URL: {url}\nparams: {params}')
             elif e.response.status_code == 429:
-                seconds = response.headers.get('Retry-After',
-                                               DEFAULT_WAITING_TIME)
-                self._log.debug(f'Too many API requests. Retrying after '
-                                f'{seconds} seconds.')
-                time.sleep(int(seconds) + 1)
-                return self._handle_response(request_type, url, params=params)
+                if retry_count < max_retries:
```

Comment:
This could be _slightly_ easier to read if the logic were inverted, i.e.:

```py
if retry_count >= max_retries:
    raise SpotifyAPIError('Maximum retries reached.')
seconds = ...
```

…then at least the "normal case" doesn't even need more reindenting.